packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -483,15 +483,9 @@ async function handler(
   if (eventType.seatsPerTimeSlot) {
     const booking = await prisma.booking.findFirst({
       where: {
-        OR: [
-          {
-            uid: rescheduleUid || reqBody.bookingUid,
-          },
-          {
-            eventTypeId: eventType.id,
-            startTime: new Date(dayjs(reqBody.start).utc().format()),
-          },
-        ],
+        uid: rescheduleUid || reqBody.bookingUid,
```

Comment:
On a second thought do we even need to search by uid here? We just need to find if there is a booking at a particular time, for which we need only eventTypeId and startTime?

When rescheduling, you will never find a booking with rescheduleUid and startTime of the new slot, so it will always be treated as first seat. Although it does not look like anything is breaking 🤔

Reply:
Yes uid is not required, but nothing will break now by having it. Since reqBody.bookingUid was used, i left it as it is.
In case of new booking, there is no bookingUid or rescheduleUId, uid will be false, prisma will remove this field from where clause, when value is false , so not a problem.
In case of reschedule, rescheduleUid itself is the bookingUid of original booking, any how the slot won't match to decide that it's second slot, so not a problem

Reply:
> On a second thought do we even need to search by uid here? We just need to find if there is a booking at a particular time, for which we need only eventTypeId and startTime?

Agree, I am not sure why `rescheduleUid` and `bookingUid` would be needed here

> When rescheduling, you will never find a booking with rescheduleUid and startTime of the new slot, since it has not been booked yet, so it will always be treated as first seat. Although it does not look like anything is breaking 🤔

Agree, also couldn't find anything that it breaks but that just makes me wondering why we even check for `isFirstSeat` 🤔

Reply:
to skip avilability check - in case of seated event (or multiple slots) , if user is trying to book second seat or (for remaining seats) then no need to check for user availability.
But need to check when trying to book for first time (or first seat)

Reply:
> to skip avilability check - in case of seated event (or multiple slots) , if user is trying to book second seat or (for remaining seats) then no need to check for user availability.

I tested rescheduling to a slot that already has seats. `isFirstSeat` was true (even though it wasn't the first seat), but the availability check didn't throw an error even when I tried to create a calendar event that would conflict

Reply:
In that case, IMO its better to remove uid from where clause.

but for fresh or new booking, if this `isFirstSeat` check is not there , booking is allowed in non-availability slots.

Reply:
> > to skip avilability check - in case of seated event (or multiple slots) , if user is trying to book second seat or (for remaining seats) then no need to check for user availability.
> 
> I tested rescheduling to a slot that already has seats. `isFirstSeat` was true (even though it wasn't the first seat), but the availability check didn't throw an error even when I tried to create a calendar event that would conflict

updated to remove `uid` in where clause, now `isFirstSeat` should be false for this scenario